lnwire/shutdown.go

```diff
@@ -85,17 +86,62 @@ func (s *Shutdown) Decode(r io.Reader, pver uint32) error {
 	// Set the corresponding TLV types if they were included in the stream.
 	if val, ok := typeMap[s.ShutdownNonce.TlvType()]; ok && val == nil {
 		s.ShutdownNonce = tlv.SomeRecordT(musigNonce)
+
+		// Remove the entry from the TLV map. Anything left in the map
+		// will be included in the custom records field.
+		delete(typeMap, s.ShutdownNonce.TlvType())
+	}
+
+	// Any records from the extra data TLV map which are in the custom
+	// records TLV type range will be included in the custom records field
+	// and removed from the extra data field.
+	customRecordsTlvMap := make(tlv.TypeMap, len(typeMap))
+	for k, v := range typeMap {
+		// Skip records that are not in the custom records TLV type
+		// range.
+		if k < MinCustomRecordsTlvType {
```

Comment:
Hmm... This makes sense. Maybe we should add this as a function or method to the custom records and use it everywhere? Then the explicit `delete` above for the nonce wouldn't be necessary anymore either.

Reply:
Or maybe the `NewCustomREcordsFromTlvTypeMap()` function should do this filtering, as we explicitly only want custom records?